pkg/autoscaler/autoscaler_server.go

```diff
@@ -129,19 +131,39 @@ func (s *AutoScaler) pollAPIServer() {
 			r := resource.NewQuantity(0, guessFormat(res))
 			r.SetMilli(want)
 			newReqs[ctr].Requests[apiv1.ResourceName(res)] = *r
-			glog.V(0).Infof("Setting %s requests[%q] = %v", ctr, res, r)
+			glog.V(4).Infof("Calculated %s requests[%q] = %v", ctr, res, r)
 		}
 		for res, cfg := range ctrcfg.Limits {
 			want := calculate(cfg, clusterSize)
 			r := resource.NewQuantity(0, guessFormat(res))
 			r.SetMilli(want)
 			newReqs[ctr].Limits[apiv1.ResourceName(res)] = *r
-			glog.V(0).Infof("Setting %s limits[%q] = %v", ctr, res, r)
+			glog.V(4).Infof("Calculated %s limits[%q] = %v", ctr, res, r)
 		}
 	}
+	if reflect.DeepEqual(s.lastReqs, newReqs) {
+		return
+	}
+
+	glog.V(0).Infof("Updating resource for nodes: %d, cores: %d",
+		clusterSize.Nodes, clusterSize.Cores)
+	logRequirements(newReqs)
 	// Update resource target with new resources.
 	if err = s.k8sClient.UpdateResources(newReqs); err != nil {
 		glog.Errorf("Update failure: %s", err)
+	} else {
+		s.lastReqs = newReqs
+	}
+}
+
+func logRequirements(reqs map[string]apiv1.ResourceRequirements) {
+	for ctr, req := range reqs {
+		for res, r := range req.Requests {
+			glog.V(0).Infof("Setting %s requests[%q] = %v", ctr, res, &r)
```

Comment:
Should we use `r` instead of `&r`? (It is a [Quantity](https://github.com/kubernetes-incubator/cluster-proportional-vertical-autoscaler/blob/31fc55a93c3c88ad4b112626245764396f1da699///vendor/k8s.io/apimachinery/pkg/api/resource/quantity.go#L88-L99) struct?)

Reply:
Tried this in testing and found that it has to be &r for a better format, i.e. to use (&r).String() method for %v printing.

Reply:
Right, seems good then